packages/components/pie-radio/test/component/pie-radio.spec.ts

```diff
@@ -17,4 +31,197 @@ test.describe('PieRadio - Component tests', () => {
+
+    test.describe('props', () => {
+        test.describe('checked', () => {
+            test('should check the radio when true', async ({ page, mount }) => {
+                // Arrange
+                await mount(PieRadio, {
+                    props: {
+                        checked: true,
+                        value: 'value',
+                    } as RadioProps,
+                    slots,
+                });
+
+                // Act
+                const radio = page.locator(inputSelector);
+
+                // Assert
+                await expect(radio).toBeChecked();
+            });
+
+            test('should not check the radio when false', async ({ page, mount }) => {
+                // Arrange
+                await mount(PieRadio, {
+                    props: {
+                        checked: false,
+                        value: 'value',
+                    } as RadioProps,
+                    slots,
+                });
+
+                // Act
+                const radio = page.locator(inputSelector);
+
+                // Assert
+                await expect(radio).not.toBeChecked();
+            });
+        });
+
+        test.describe('disabled', () => {
+            test('should enable the radio when false', async ({ mount, page }) => {
+                // Arrange
+                await mount(PieRadio, {
+                    props: {
+                        disabled: false,
+                        value: 'value',
+                    } as RadioProps,
+                    slots,
+                });
+
+                // Act
+                const radio = page.locator(inputSelector);
+
+                // Assert
+                await expect(radio).not.toBeDisabled();
+            });
+
+            test('should disable the radio when true', async ({ mount, page }) => {
+                // Arrange
+                await mount(PieRadio, {
+                    props: {
+                        disabled: true,
+                        value: 'value',
+                    } as RadioProps,
+                    slots,
+                });
+
+                // Act
+                const radio = page.locator(inputSelector);
+
+                // Assert
+                await expect(radio).toBeDisabled();
+            });
+        });
+
+        test.describe('required', () => {
+            test('should not add a required attribute if the prop is not provided', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieRadio, {
+                    props: {
+                        value: 'value',
+                    } as RadioProps,
+                });
+
+                // Act
+                const radio = component.locator(inputSelector);
+
+                // Assert
+                expect((await radio.getAttribute('required'))).toBe(null);
+            });
+
+            test('should apply the required attribute to the input element', async ({ mount }) => {
```

Comment:
Would it be possible to test that `valueMissing` is `true` instead of checking the attribute has been rendered? (assuming this is the expected ValidityState)

Reply:
Another good idea. That validity state only applies when there are multiple radios in the named group, and I'll be adding more tests like those in the next PR.